### PR TITLE
feat: add deployment envs and start script

### PR DESCRIPTION
**Pull request by theyokohamalife**
Part of #29 

# What changed
1. Creates `yarn start` to be utilized as the run command in the deployment process.
2. Adds the `typeRoots` to tsconfig  so it knows where to find the types for the environment variables.
3. Changes the value of `port` so it reads from the .env file if it exists. This will automatically be set by the app platform on Digital Ocean. ([documentation](https://www.digitalocean.com/community/tutorials/how-to-build-a-graphql-api-with-prisma-and-deploy-to-digitalocean-s-app-platform))
4. Updates the .env.sample file to include the `ENV` and `SERVER_PORT`

# Testing instructions
1. Update your .env file so it includes the new fields added to the .env.sample
2. Add the following to the bottom of index.ts and run `yarn start`:

```
console.log(`SERVER PORT: ${process.env.SERVER_PORT}`)
console.log(`ENV: ${process.env.ENV}`)
```

### CODE REVIEW

**Comment by theyokohamalife** [comment]
@socketsecurity ignore @apollo/protobufjs@1.2.7

File: .env.sample [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # .env.sample 
 # Please make a copy of this file to `.env` and update the POSTGRES_PASSWORD
 # and other secrets
+ENV=test
```
Comment (ann-kilzer) [inline .env.sample:4]: This seems redundant with `NODE_ENV` defined below

File: package.json [hunk 2]
```diff
@@ -13,7 +13,8 @@
         "lint:check": "eslint . --ext .js,.ts,.json",
         "lint:ci": "eslint . --ext .js,.ts,.json",
         "migrate": "prisma migrate dev",
-        "migrate:deploy": "prisma migrate deploy"
+        "migrate:deploy": "prisma migrate deploy",
+        "start": "yarn && ts-node --transpile-only src/index.ts"
```
Comment (ann-kilzer) [inline package.json:17]: 1. Let's avoid calling yarn inside yarn as it does weird things
2. `start` and `dev` are pretty ambiguous. Please merge #124 first and let's revisit this

**Comment by ann-kilzer** [comment]
Thanks for looking into this. Left some initial feedback

**Comment by theyokohamalife** [comment]
@ann-kilzer  Please ignore this PR for now. It's still a draft and not ready for review. At this point I'm just using it to familiarize myself with Digital Ocean.

**Pull request closed without merging**